python/lsst/pipe/analysis/fakesAnalysis.py

```diff
@@ -0,0 +1,615 @@
+# This file is part of pipe analysis
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Make QA plots for data with fake sources inserted
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from collections import defaultdict
+from matplotlib import colors, gridspec
+from matplotlib.path import Path
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+from scipy.interpolate import UnivariateSpline
+
+from lsst.qa.explorer.match import match_lists
+
+__all__ = ["addDegreePositions", "matchCatalogs", "addNearestNeighbor", "sigmaMAD", "getPlotInfo",
+           "areaDepth", "positionCompare"]
+
+
+def addDegreePositions(catalog, raCol, decCol):
+    """Add RA and Dec columns in degrees, requires input columns in radians.
```

Comment:
This does still assume the inputs are in radians... could you add an assert or something to verify that the `catalog[raCol].values` and `catalog[decCol].values` all fall within whatever reasonable radian range one would expect? So, for instance, if an [ra, dec] pair starts off as [150, -29], it would raise an exception because those are not reasonable radian values and the coordinate is probably already in degrees.

Reply:
Asserts checking if the ra and dec values are between 0 and 2pi have been added.